src/_P039_Thermocouple.ino

```diff
@@ -484,12 +751,868 @@ int Plugin_039_convert_two_complement(uint32_t value, int nr_bits) {
+      }
+    }
+
+  # endif // ifndef BUILD_NO_DEBUG
+
+  bool ValueValid = false;
+
+  if (registers[MAX31865_FAULT] == 0x00)
+    ValueValid = true;
+
+  # ifndef BUILD_NO_DEBUG
+
+    if (loglevelActiveFor(LOG_LEVEL_DEBUG))
+    {
+      String log;
+      log.reserve(66u);
+      log = F("P039 : Temperature :");
+      log += F(" registers[MAX31865_FAULT]: ");
+      log += String(registers[MAX31865_FAULT], HEX);
+      log += F(" ValueValid: ");
+      log += String(ValueValid, BIN);
+      addLog(LOG_LEVEL_DEBUG, log);
+    }
+
+  # endif // ifndef BUILD_NO_DEBUG
+
+  if (ValueValid)
+  {
+    rawValue >>= 1; // bottom fault bits is unused
+
+    float temperature = Plugin_039_convert_to_temperature(rawValue, getNomResistor(P039_RTD_TYPE), P039_RTD_RES);
+
+    # ifndef BUILD_NO_DEBUG
+
+      if (loglevelActiveFor(LOG_LEVEL_DEBUG))
+      {
+        String log;
+        log.reserve(66u);
+        log = F("P039 : Temperature :");
+        log += F(" rawValue: ");
+        log += String(rawValue, DEC);
+        log += F(" temperature: ");
+        log += String(temperature, DEC);
+        log += F(" P039_RTD_TYPE: ");
+        log += String(P039_RTD_TYPE, DEC);
+        log += F(" P039_RTD_RES: ");
+        log += String(P039_RTD_RES, DEC);
+        addLog(LOG_LEVEL_DEBUG, log);
+      }
+
+    # endif // ifndef BUILD_NO_DEBUG
+
+    // add offset handling from configuration webpage
+    temperature += P039_RTD_OFFSET;
+
+    // Calculate Celsius
+    return temperature;
+  }
+  else
+  {
+    // Fault state, thus output no value.
+    return NAN;
+  }
+}
+
+void clearFaults(uint8_t l_CS_pin_no)
+{
+  uint8_t l_reg = 0u;
+
+  // read in config register
+  l_reg = read8BitRegister(l_CS_pin_no,(MAX31865_READ_ADDR_BASE + MAX31865_CONFIG));
+
+ 
+  // clear all faults ( write "0" to D2, D3, D5; write "1" to D2)
+  l_reg &= ~(MAX31865_SET_ONE_SHOT | MAX31865_FAULT_CTRL_MASK);
+  l_reg |= MAX31865_CLEAR_FAULTS;
+
+  // write configuration to MAX31865 to enable VBIAS
+  write8BitRegister(l_CS_pin_no, (MAX31865_WRITE_ADDR_BASE + MAX31865_CONFIG), l_reg);
+
+}
+
+void handleBias(uint8_t l_CS_pin_no, bool l_active)
+{
+  uint8_t l_reg = 0u;
+
+  // read in config register
+  l_reg = read8BitRegister(l_CS_pin_no, (MAX31865_READ_ADDR_BASE + MAX31865_CONFIG));
+
+  if (l_active)
+  {
+    // activate BIAS
+    l_reg |= MAX31865_SET_VBIAS_ON;
+  }
+  else
+  {
+    // deactivate BIAS
+    l_reg &= ~MAX31865_SET_VBIAS_ON;
+  }
+
+  // write configuration to MAX31865 to enable VBIAS
+  write8BitRegister(l_CS_pin_no, (MAX31865_WRITE_ADDR_BASE + MAX31865_CONFIG), l_reg);
+
+}
+
+void chooseFilterType(uint8_t l_CS_pin_no, uint8_t l_filtType)
+{
+  uint8_t l_reg = 0u;
+
+  // read in config register
+  l_reg = read8BitRegister(l_CS_pin_no, (MAX31865_READ_ADDR_BASE + MAX31865_CONFIG));
+
+  // configure access to sensor (2/4-wire OR 3 wire)
+  switch (l_filtType)
+  {
+    case 0:
+              l_reg &= ~MAX31865_SET_50HZ;
+              break;
+    case 1:
```

Comment:
This kind of 'case' indentation is rather unusual.
You could add the Uncrustify add-on in VSCode (instructions are in the add-on documentation when selecting it in VSCode). After installation the Format Document option will use the provided uncrustify.cfg file, included in the source, to apply 'standardized' formatting. It may also add missing curly braces where appropriate, and other readability improvements.

Reply:
Thats how we did it - back then, when i have still been active in daily coding. I will install the Uncrustify add-on and apply it.
Any supporting tool is highly welcome !

Reply:
You could have a look at the required and optional plugins for VSCode [here](https://espeasy.readthedocs.io/en/latest/Participate/PlatformIO.html?highlight=optional#install) (also talks about Uncrustify).

Reply:
Thank You for the hint - gives some answers to future potential questions ! Reading the documentation is the key...

Reply:
I told my girlfriend a while ago that our ReadTheDocs page was visited by over 20'000 unique visitors from over 100 countries.
So my conclusion was that we must have roughly over 20'000 active users.
Her reply:
Nope, you've got like 20'000 female users, because men don't read manuals :)

Reply:
Um, i did prove her right...

Reply:
Uncrustify has been applied to the code - it seems there was not much change on the case indentation. I will check if something goes wrong with the tool set up.

Reply:
I think the main issue could be the missing  pair of `{ ...}` around each of the cases. I will insert them in one of my next pushes. Partly added already now for some switch-case sections.